sycl/include/sycl/multi_ptr.hpp

```diff
@@ -126,8 +136,8 @@ class multi_ptr {
   multi_ptr(accessor<ElementType, Dimensions, Mode, access::target::device,
                      isPlaceholder, PropertyListT>
                 Accessor)
-      : multi_ptr(
-            detail::cast_AS<decorated_type *>(Accessor.get_pointer().get())) {}
+      : multi_ptr(detail::cast_AS<decorated_type *>(
+            Accessor.template get_multi_ptr<access::decorated::yes>().get())) {}
```

Comment:
Do we even need the `.get()` call here? Do we not always have an appropriate conversion or copy ctor?

Reply:
Without that `.get()` reinterpret_cast from 'sycl::multi_ptr<T>` to `T*` fails to compile.

Reply:
Oh, sorry, I was not being specific enough. We would also need to remove the `cast_AS`, i.e. this would end up being

```
            :multi_ptr(Accessor.template get_multi_ptr<DecorateAddress>()) {}
```

Reply:
@steffenlarsen, thanks, done.

Reply:
thanks @steffenlarsen , done.